crates/uv/src/commands/pip/latest.rs

```diff
@@ -33,6 +33,86 @@ impl LatestClient<'_> {
     ) -> anyhow::Result<Option<DistFilename>, uv_client::Error> {
         debug!("Fetching latest version of: `{package}`");
 
+        let exclude_newer = self.exclude_newer.exclude_newer_package(package);
+
+        let mut latest: Option<DistFilename> = None;
+
+        let consider_candidate = |filename: &DistFilename, file: &File| {
```

Comment:
Can we make this a separate function? You can pass in an `impl Iterator<...>` if that helps